ConsoleApp7/Window.cs

```diff
@@ -183,135 +211,111 @@ bool IsCollided(Sphere sphere1, Sphere sphere2)
 
 
 
-        public void ProcessPhysics(double dt)
+        public void ProcessPhysics(double dt, int i, int k)
         {
             for (int r = 0; r < AccuracyLim; r++)
-            {                
-                GravityAndBoundingBox(dt);
-                
-                //Sphere2SphereCollisionDetermine();
-                //Sphere2SphereCollisionCalculate();                
+            {
+                if (SimActive % 2 == 1)
+                {
+                    GravityAndBoundingBox(dt);
```

Comment:
Try to split it into some phases and implement simple narrow and broad phase algorithms (pcm and mbp for example). So you can optimize processing due to optimizing collision detection and it's solving. Check example in "Game Physics" by David H. Eberly